src/contract.rs

```diff
@@ -244,3 +293,53 @@ impl From<&Nft> for IoNFT {
         }
     }
 }
+
+impl From<&Contract> for State {
+    fn from(value: &Contract) -> Self {
+        let Contract {
+            token,
+            token_id,
+            owner,
+            transactions,
+            collection,
+            constraints,
+        } = value;
+
+        let owners = token
+            .owner_by_id
+            .iter()
+            .map(|(hash, actor_id)| (*actor_id, *hash))
+            .collect();
+
+        let transactions = transactions
+            .iter()
+            .map(|(hash, event)| (*hash, event.clone()))
+            .collect();
+
+        let token_metadata_by_id = token
+            .token_metadata_by_id
+            .iter()
+            .map(|(id, metadata)| {
+                let metadata = metadata.as_ref().unwrap();
+                let nft = Nft {
+                    owner: *token.owner_by_id.get(id).unwrap(),
```

Comment:
`get` + `unwrap` often can be replaced by `[]`
```suggestion
                    owner: *token.owner_by_id[id],
```